Doc/constraints.txt

```diff
@@ -0,0 +1,24 @@
+# We have upper bounds on our transitive dependencies here
+# To avoid new releases unexpectedly breaking our build.
+# This file can be updated on an ad-hoc basis,
+# though it will probably have to be updated
+# whenever Doc/requirements.txt is updated.
+
+# Direct dependencies of Sphinx
+babel<3
+colorama<0.5
+imagesize<1.5
+Jinja2<3.2
```

Comment:
```suggestion
```
this contraint is superfluous; there's `jinja2==3.0.3` in `requirements.txt`

Reply:
This is a backport, minimal changes are preferable.